Fix: Linkgraph legend assumes strings are small.

PeterN opened the pull request:
## Motivation / Problem

The linkgraph legend is designed for all legend strings to be in the small font, but doesn't enforce this.

![image](https://user-images.githubusercontent.com/639850/235153413-c411c498-e420-4d42-9889-ba2dbde3a3dd.png)

## Description

Ensure all linkgraph legend strings are drawn with FS_SMALL forced, so that it does not rely on NewGRF authors.

![image](https://user-images.githubusercontent.com/639850/235153856-1bd7f016-daa1-466c-8de5-2be0932cdd6f.png)

## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

2TallTyler reviewed (commented):
Is it worth removing the string codes from the vanilla strings? (Could be a separate PR)

PeterN committed:
Change: Rename text colour NWidgetPart to SetTextStyle and add font size.

This allows to set both text colour and size for any widget with default
drawing.

PeterN committed:
Fix: Linkgraph legend assumes strings are small.

Ensure all linkgraph legend strings are drawn with FS_SMALL forced, so
that it does not rely on NewGRF authors.

PeterN committed:
Fix: Force cargo abbreviation to small font in station list.

PeterN committed:
Change: Remove {TINY_FONT} from cargo abbreviation strings for English.

PeterN committed:
Change: Remove {TINY_FONT} from cargo abbreviation strings for other translations.

glx22 reviewed (dismissed):
Not sure how EINTS will react for translations editions (worst case it will revert or remove the strings)

glx22 commented on src/widget.cpp:
```diff
@@ -487,14 +487,14 @@ static inline void DrawImageButtons(const Rect &r, WidgetType type, Colours colo
  * @param str     Text to draw.
  * @param align   Alignment of the text.
  */
```
Missing doxygen for `fs`

glx22 commented on src/widget.cpp:
```diff
@@ -504,11 +504,11 @@ static inline void DrawLabel(const Rect &r, WidgetType type, bool clicked, TextC
  * @param str    Text to draw.
  * @param align  Alignment of the text.
  */
```
Same

glx22 commented on src/widget.cpp:
```diff
@@ -519,10 +519,10 @@ static inline void DrawText(const Rect &r, TextColour colour, StringID str, Stri
  * @param str         Text to draw.
  * @param align       Alignment of the text.
  */
```
And here too

glx22 commented on src/widget.cpp:
```diff
@@ -673,11 +673,11 @@ static inline void DrawHorizontalScrollbar(const Rect &r, Colours colour, bool l
  * @param str         Text of the frame.
  * @param align       Alignment of the text in the frame.
  */
```
And here

glx22 commented on src/widget.cpp:
```diff
@@ -802,7 +802,7 @@ static inline void DrawCloseBox(const Rect &r, Colours colour)
  * @param str         Text to draw in the bar.
  * @param align       Alignment of the text.
  */
```
Again
